packages/http/src/mocker/HttpMocker.ts

```diff
@@ -33,24 +33,36 @@ class HttpMocker
     const payloadGenerator: PayloadGenerator =
       config && typeof config.mock !== 'boolean' && config.mock.dynamic ? generate : generateStatic;
 
-    return pipe(
-      withLogger(logger => {
-        // setting default values
-        const acceptMediaType = input.data.headers && caseless(input.data.headers).get('accept');
-        config = config || { mock: false, validateRequest: true, validateResponse: true };
-        const mockConfig: IHttpOperationConfig =
-          config.mock === false ? { dynamic: false } : Object.assign({}, config.mock);
-
-        if (!mockConfig.mediaTypes && acceptMediaType) {
-          logger.info(`Request contains an accept header: ${acceptMediaType}`);
-          mockConfig.mediaTypes = acceptMediaType.split(',');
-        }
-
-        return mockConfig;
-      }),
-      chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
-      chain(result => assembleResponse(result, payloadGenerator)),
-    );
+    const isHead = input.data.method === 'head';
+    const r = get(resource, ['responses', '0'], { code: 200, headers: [] });
```

Comment:
This should only be happening if `isHead` is true, otherwise the result will never be used.

Reply:
This won't be enough, it's simply relying on the first response; imagine my first response is a 500 instead of a 200 (or also a 204) — this is not what we want to return. We still need to go through some parts of the negotiation process. It would be a good idea to add a test to make sure it behaves correctly on this.

1. We still need input validation (query string and whatever)
2. We need to still go through the regular response checking.

I guess I was kind of wrong when I was saying we can skip most of the negotiation process, I am sorry.

I think the best way to plug the code would be here:

https://github.com/stoplightio/prism/blob/master/packages/http/src/mocker/negotiator/NegotiatorHelpers.ts#L140

We can detect here if we're dealing with an `head` request and in such case early return a right with an `empty` body skipping all the code we see. If we could write a log line as well, that'd be great.

Does that make sense?